### PR TITLE
feat: add put consents/{ID} handler

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: src/server/handlers/consents/{ID}.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+async function put(_context: any, request: Request, h: ResponseToolkit): Promise<ResponseObject> {
+  const span = (request as any).span
+  // Trust that hapi parsed the ID and Payload for us
+  const consentsRequestId: string = request.params.ID
+  const payload = request.payload as types.ConsentsIDPayload
+
+  try {
+    const tags: { [id: string]: string } = getSpanTags(
+      request,
+      // todo: add a consent-request/thirdparty? eventType to central-services-shared
```
Comment (lewisdaly) [inline src/server/handlers/consents/{ID}.ts:54]: Can we please make a ticket for this/link to an existing ticket?

File: test/unit/server/handlers/consents/{ID}.test.ts [hunk 2]
```diff
@@ -66,47 +81,47 @@ describe('consentsIdGenerateChallenge handler', () => {
...
-      expect(mockForwardConsentsIdGenerateChallengeRequest).toHaveBeenCalledWith(...expected)
+      expect(mockForwardConsentsIdPutRequest).toHaveBeenCalledWith(...expected)
     })
 
     it('handles validation errors synchronously', async () => {
       // Arrange
       const request = {
-        ...mockPostConsentsIdGenerateChallengeRequest,
+        ...mockPostConsentsIdPutRequest,
         // Will setting the span to null do stuff?
```
Comment (lewisdaly) [inline test/unit/server/handlers/consents/{ID}.test.ts:118]: Can we move this copypaste?

File: test/unit/server/handlers/consents/{ID}/generateChallenge.test.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      await new Promise(resolve => setImmediate(resolve))
+      // The main test here is that there is no unhandledPromiseRejection!
+      expect(mockForwardConsentsIdGenerateChallengeRequest).toHaveBeenCalledWith(...expected)
+    })
+
+    it('handles validation errors synchronously', async () => {
+      // Arrange
+      const request = {
+        ...mockPostConsentsIdGenerateChallengeRequest,
+        // Will setting the span to null do stuff?
```
Comment (lewisdaly) [inline test/unit/server/handlers/consents/{ID}/generateChallenge.test.ts:103]: nit: can we please remove this copy paste?

**Review by lewisdaly — CHANGES_REQUESTED** [review]
Looking good! Just a couple minor points to tidy and clarify.

File: test/unit/server/handlers/consents/{ID}/generateChallenge.test.ts (hunk 3 above)
Comment (kleyow) [reply test/unit/server/handlers/consents/{ID}/generateChallenge.test.ts:103]: let me leave this for another PR.

**Review by sridharvoruganti — APPROVED** [review]
lgtm

File: src/server/handlers/consents/{ID}.ts (hunk 1 above)
Comment (kleyow) [reply src/server/handlers/consents/{ID}.ts:54]: I've made the ticket due to your previous comment on another PR. https://github.com/mojaloop/project/issues/1750

**Pull request merged**